src/Components/Homepage.js

```diff
@@ -1,38 +1,58 @@
-import React, {Component} from 'react';
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+
 import Header from './Header';
 import Main from './Main';
 import Footer from './Footer';
 import ErrorBoundary from './ErrorBoundary';
-import './Homepage.css';
 import { MovieModal } from "./MovieModal";
+import { MovieInfo } from "./MovieInfo";
 
-class Homepage extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isAddMovieModalShown: false
-        }
-    }
+import './Homepage.css';
+
+export const Homepage = () => {
+    const useToggle = (isComponentShown = false) => {
```

Comment:
I think that you can move it in separate file. After this you can just import it and don't pass through props

Reply:
fixed